doc/lsst.source.injection/reference/1_make_injection_pipeline.rst

```diff
@@ -0,0 +1,167 @@
+   * - deepCoadd
+     - inject_coadd.yaml_
+     - :lsst-task:`~lsst.source.injection.CoaddInjectTask`
+
+.. _inject_exposure.yaml: https://github.com/lsst/source_injection/blob/main/pipelines/inject_exposure.yaml
+.. _inject_visit.yaml: https://github.com/lsst/source_injection/blob/main/pipelines/inject_visit.yaml
+.. _inject_coadd.yaml: https://github.com/lsst/source_injection/blob/main/pipelines/inject_coadd.yaml
+
+.. _lsst.source.injection-reference-make-injection-pipeline-command-line:
+
+Making an Injection Pipeline on the Command Line
+================================================
+
+The :doc:`make_injection_pipeline <../scripts/make_injection_pipeline>` command line script is used to generate a complete source injection pipeline definition YAML file.
+More information on the operation of this script may be obtained by running ``make_injection_pipeline --help``.
+
+An example calling this script from the command line:
+
+.. code-block:: shell
+
+    $SOURCE_INJECTION_DIR/bin/make_injection_pipeline \
+    -t postISRCCD \
+    -r $DRP_PIPE_DIR/pipelines/HSC/DRP-RC2.yaml \
+    -i $SOURCE_INJECTION_DIR/pipelines/inject_exposure.yaml \
+    -f my_injection_pipeline.yaml
+
+*where*
+
+    `$DRP_PIPE_DIR`
+        The path to the `drp_pipe` package directory.
+
+    `$SOURCE_INJECTION_DIR`
+        The path to the source injection package directory.
+
+The above command will save a complete and fully expanded pipeline definition file into the file ``my_injection_pipeline.yaml``.
+In this example, synthetic sources are to be injected into the ``postISRCCD`` dataset type, using the ``HSC/DRP-RC2.yaml`` pipeline definition file as a reference.
+As the ``postISRCCD`` dataset type has dimensions of ``exposure``, the ``inject_exposure.yaml`` source injection pipeline definition file stub is used which contains the :lsst-task:`~lsst.source.injection.ExposureInjectTask` task.
+
+.. note::
+    To print the fully qualified output pipeline to the terminal window instead of saving it to a file, omit the ``-f`` option in the above example.
+
+.. _lsst.source.injection-reference-make-injection-pipeline-python:
+
+Making an Injection Pipeline in Python
+======================================
+
+The :py:func:`~lsst.source.injection.make_injection_pipeline` Python function is used to generate a complete source injection pipeline definition YAML file in Python:
+
+.. code-block:: python
+
+    from lsst.source.injection import make_injection_pipeline
+
+More information on the operation of this function may be obtained by running ``make_injection_pipeline?`` in a Python interpreter.
+
+An example calling this function in Python:
+
+.. code-block:: python
+
+    # Construct the Pipeline object.
+    pipeline = make_injection_pipeline(
+        dataset_type_name="postISRCCD",
+        reference_pipeline="$DRP_PIPE_DIR/pipelines/HSC/DRP-RC2.yaml",
+        injection_pipeline="$SOURCE_INJECTION_DIR/pipelines/inject_exposure.yaml",
```

Comment:
Similar comment to above - no butler is currently required here. Adding the automatic inference of the `injection_pipeline` would require that to exist on-disk already.

Another side note: the user may be querying against a completely new repo where `postISRCCD` has not yet been registered. In such a scenario, automatic inference of the injection pipeline would fail. This may be an edge case, but also something to consider.

Reply:
See above, :point_right: [DM-39748](https://jira.lsstcorp.org/browse/DM-39748)